src/plugins/plugin.legend.js

```diff
@@ -117,7 +139,272 @@ export class Legend extends Element {
       legendItems.reverse();
     }
 
-    this.legendItems = legendItems;
+    this.navigation.legendItems = this.legendItems = legendItems;
```

Comment:
Personally I think doing this in 2 steps is a lot more readable and less error prone:
```suggestion
    this.legendItems = legendItems;
    this.navigation.legendItems = legendItems;
```

Reply:
I agree. Solved.